integrations/opensearch/src/haystack_integrations/components/retrievers/open_search_hybrid_retriever.py

```diff
@@ -0,0 +1,322 @@
+# SPDX-FileCopyrightText: 2023-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, Dict, List, Literal, Optional, Union
+
+from haystack import Pipeline, default_from_dict, default_to_dict, logging
+from haystack.components.embedders import SentenceTransformersTextEmbedder
+from haystack.components.joiners import DocumentJoiner
+from haystack.components.joiners.document_joiner import JoinMode
+from haystack.document_stores.types import FilterPolicy
+from haystack.lazy_imports import LazyImport
+from haystack.utils import ComponentDevice
+
+from haystack_integrations.components.retrievers.opensearch import OpenSearchBM25Retriever, OpenSearchEmbeddingRetriever
+from haystack_integrations.document_stores.opensearch import OpenSearchDocumentStore
+
+logger = logging.getLogger(__name__)
+
+# Use LazyImport to conditionally import haystack super_component, which is only available in newer versions
+with LazyImport(
+    "To use the OpenSearchHybridRetriever you need more recent version of haystack. Run 'pip install haystack-ai>=2.13.0'"  # noqa: E501
+) as haystack_imports:
+    from haystack import super_component
+
+# Trigger an error message when the OpenSearchHybridRetriever is imported without haystack-ai>=2.13.0
+haystack_imports.check()
+
+
+@super_component
+class OpenSearchHybridRetriever:
+    """
+    A hybrid retriever that combines embedding-based and keyword-based retrieval from OpenSearch.
+
+    This component requires haystack-ai>=2.13.0 to work properly.
+    """
+
+    def __init__(
+        self,
+        document_store: OpenSearchDocumentStore,
+        *,
+        # SentenceTransformer
+        text_embedder_model: str = "sentence-transformers/all-mpnet-base-v2",
+        device: Optional[ComponentDevice] = None,
+        normalize_embeddings: bool = False,
+        model_kwargs: Optional[Dict[str, Any]] = None,
+        tokenizer_kwargs: Optional[Dict[str, Any]] = None,
+        config_kwargs: Optional[Dict[str, Any]] = None,
+        encode_kwargs: Optional[Dict[str, Any]] = None,
+        backend: Literal["torch", "onnx", "openvino"] = "torch",
+        # OpenSearchBM25Retriever
+        filters_bm25: Optional[Dict[str, Any]] = None,
+        fuzziness: Union[int, str] = "AUTO",
+        top_k_bm25: int = 10,
+        scale_score: bool = False,
+        all_terms_must_match: bool = False,
+        filter_policy_bm25: Union[str, FilterPolicy] = FilterPolicy.REPLACE,
+        custom_query_bm25: Optional[Dict[str, Any]] = None,
+        # OpenSearchEmbeddingRetriever
+        filters_embedding: Optional[Dict[str, Any]] = None,
+        top_k_embedding: int = 10,
+        filter_policy_embedding: Union[str, FilterPolicy] = FilterPolicy.REPLACE,
+        custom_query_embedding: Optional[Dict[str, Any]] = None,
+        # DocumentJoiner
+        join_mode: Union[str, JoinMode] = JoinMode.CONCATENATE,
```

Comment:
I'd say `RECIPROCAL_RANK_FUSION` is a better default for a hybrid retriever because scores of the OpenSearchBM25Retriever are not scaled to [0,1] by default and thus are not comparable to the scores of the OpenSearchEmbeddingRetriever.